src/fu-util.c

```diff
@@ -2021,6 +2029,64 @@ fu_util_downgrade (FuUtilPrivate *priv, gchar **values, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_util_reinstall (FuUtilPrivate *priv, gchar **values, GError **error)
+{
+	const gchar *remote_id;
+	FwupdRelease *rel = NULL;
+	g_autoptr(GPtrArray) rels = NULL;
+	g_autoptr(FwupdDevice) dev = fu_util_get_device_or_prompt (priv,
+								   values,
+								   error);
+	if (dev == NULL)
+		return FALSE;
+
+	/* try to lookup/match release from client */
+	rels = fwupd_client_get_releases (priv->client, fwupd_device_get_id (dev),
+					  NULL, error);
+	if (rels == NULL)
+		return FALSE;
+	for (guint j = 0; j < rels->len; j++) {
+		FwupdRelease *rel2 = g_ptr_array_index (rels, j);
+		if (fu_common_vercmp (fwupd_release_get_version (rel2),
+		    fu_device_get_version (dev)) == 0) {
+			rel = rel2;
```

Comment:
I'd feel a lot more warm-and-fuzzy if this was a g_object_ref() and the toplevel rel was g_autoptr.